index.js

```diff
@@ -191,3 +205,11 @@ function zOrder(x, y) {
 
     return x | (y << 1);
 }
+
+function hasAllKeys(tags, requiredLookup, requiredCount) {
+    var found = 0;
+    for (var i = 0, j = tags.length; i < j; i += 2) {
```

Comment:
You don't have to cache the length in for loops in modern JS engines. This is optimized automatically.